tests/Unit/SymfonyCache/EventDispatchingHttpCacheTest.php

```diff
@@ -142,6 +144,16 @@ public function testAbortInvalidate()
     }
 }
 
+class AppCache extends HttpCache implements CacheInvalidationInterface
+{
+    use EventDispatchingHttpCache;
+
+    public function fetch(Request $request, $catch = false)
+    {
+        return $this->fetch($request, $catch);
```

Comment:
if this is not creating an endless loop, you probably need to do the same in the other HttpCache above